Dockerfile.e2e-tests

```diff
@@ -0,0 +1,23 @@
+# By default we pin to amd64 sha. Use make docker to automatically adjust for arm64 versions.
+ARG BASE_DOCKER_SHA="14d68ca3d69fceaa6224250c83d81d935c053fb13594c811038c461194599973"
+
+# Taking a non-alpine image for e2e tests so that cgo can be enabled
+FROM golang:1.17 as builder
+
+WORKDIR $GOPATH/src/github.com/thanos-io/thanos
+# Change in the docker context invalidates the cache so to leverage docker
+# layer caching, moving update and installing apk packages above COPY cmd
+# More info https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache
```

Comment:
Looks like this was just copy pasted from `Dockerfile.multi-stage` and we don't need it here?

Reply:
Yup, actually I wanted to keep the next set of comments here, which were related to the "COPY" command, I guess I've mistakenly removed that and kept this comment, no problems! I'll make the change